app/controllers/api/memberships_controller.rb

```diff
@@ -1,14 +1,12 @@
 class API::MembershipsController < API::RestfulController
 
-  def create
-    params[:group_id] = params[:membership][:group_id]
-    load_and_authorize :group
-    event = MembershipService.join_group group: @group, user: current_user
+  def join_group
+    @group = Group.find(params[:group_id])
```

Comment:
We're running into that weird thing again where we load outside the service, then authorize inside it.

I think we talked about making the services smart enough to take an id, but I'm not sure if that feels too great either.

```
class MembershipService
  def join_group(group_id: nil, group: nil, actor:, params:)
    group ||= ModelLocator.find group_id
    actor.ability.authorize! :join_group, group
    # etc..
  end
end
```

I dunno if that feels any better, actually.

At the very least I think I'd prefer using ModelLocator here so you can pass an id or a key.
```
@group = ModelLocator.new(:group, params).locate
```